### PR TITLE
[ONNX] Fix ONNX aten::mul exporter with boolean inputs

### PR DESCRIPTION
**Pull request by wschin**
Continue work left in #72102.

The current exporter always export `aten::mul` to ONNX `Mul`. However, ONNX `Mul` [doesn't support Boolean](https://github.com/onnx/onnx/blob/main/docs/Operators.md#type-constraints-92) so we need to explicitly use ONNX `And` in this case.

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -373,7 +373,10 @@ def rsub(g, self, other, alpha=None):
 
 
 def mul(g, self, other):
-    return g.op("Mul", self, other)
+    if symbolic_helper._is_bool(self) and symbolic_helper._is_bool(other):
+        return g.op("And", self, other)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:377]: Please add comment explaining this is due to ONNX Mul does not support boolean type inputs

Comment (wschin) [reply torch/onnx/symbolic_opset9.py:378]: Ok

File: torch/onnx/symbolic_helper.py [hunk 2]
```diff
@@ -535,23 +536,35 @@ def _slice_helper(g, input, axes, starts, ends, steps=None, dynamic_slice=False)
...
+    "_ScalarAndTensorElementTypeGroup", ("tensor_element_types", "scalar_types")
+)
+_FPTypeGroup = _ScalarAndTensorElementTypeGroup(
+    (torch.float16, torch.float32, torch.float64, torch.bfloat16),
+    ("Float", "Double", "Half", "BFloat16"),
+)
+_BoolTypeGroup = _ScalarAndTensorElementTypeGroup((torch.bool,), ("Bool",))
+
+
+def _is_in_type_group(value, type_set):
```
Comment (BowenBao) [inline torch/onnx/symbolic_helper.py:549]: FYI, @thiagocrepaldi is working on the same part of code. Will need some coordination when merging these PRs.

**Review by BowenBao — APPROVED** [review]

File: torch/onnx/symbolic_helper.py (hunk 2 above)
Comment (wschin) [reply torch/onnx/symbolic_helper.py:549]: Can we merge this first? This bug has been found for more than half year..

Comment (thiagocrepaldi) [reply torch/onnx/symbolic_helper.py:549]: Indeed my iz changes `_if_in_fp` while this PR just put its implementation on a helper. Go ahead and merge and I can rebase after it is in

**Review by thiagocrepaldi — APPROVED** [review]

File: torch/onnx/symbolic_helper.py (hunk 2 above)
Comment (justinchuby) [reply torch/onnx/symbolic_helper.py:549]: I find myself working on this part of the code too in #81953. @thiagocrepaldi let's chat tomorrow?

**Pull request closed without merging**